go/vt/vtgate/planbuilder/select.go

```diff
@@ -242,11 +242,27 @@ func createSelectOperator(ctx *plancontext.PlanningContext, selStmt sqlparser.Se
 }
 
 func isOnlyDual(sel *sqlparser.Select) bool {
-	if sel.Where != nil || sel.GroupBy != nil || sel.Having != nil || sel.Limit != nil || sel.OrderBy != nil {
+	if sel.Where != nil || sel.GroupBy != nil || sel.Having != nil || sel.OrderBy != nil {
 		// we can only deal with queries without any other subclauses - just SELECT and FROM, nothing else is allowed
 		return false
 	}
 
+	if sel.Limit != nil {
+		if sel.Limit.Offset != nil {
+			return false
+		}
+		limit := sel.Limit.Rowcount
+		switch limit := limit.(type) {
+		case nil:
+			return true
```

Comment:
Is it correct here that nothing below is further checked? 

Like the table name checks?

Reply:
you are absolutely right! stupid me...